addons/l10n_cl/models/account_move.py

```diff
@@ -40,10 +44,51 @@ def _get_l10n_latam_documents_domain(self):
+                                                'document. Please set the current tax payer type of this customer'))
+                if (tax_payer_type == '4' or country_id != self.env.ref('base.cl')) and \
+                        latam_document_type_code not in ['110', '111', '112']:
+                    raise ValidationError(_('Document types for foreign customers must be export type'))
+            if rec.journal_id.type == 'sale' and not rec.journal_id.l10n_latam_use_documents:
+                raise ValidationError(_('In Chilean localization, you need to use documents in the sales journal'
+                                        'in order to register your sales properly'))
+            if rec.journal_id.type == 'purchase' and rec.journal_id.l10n_latam_use_documents:
+                if vat != '60805000-0' and latam_document_type_code == '914':
+                    raise ValidationError(_('The DIN document is intended to be used only with RUT 60805000-0'
+                                            ' (Tesorería General de La República)'))
+                if not tax_payer_type or not vat:
+                    if country_id == self.env.ref('base.cl') and latam_document_type_code not in [
+                            '35', '38', '39', '41']:
+                        raise ValidationError(_('Tax payer type and vat number are mandatory for this type of '
+                                                'document. Please set the current tax payer type of this supplier'))
+                if tax_payer_type == '2' and latam_document_type_code not in ['70', '71', '56', '61']:
+                    raise ValidationError(_('The tax payer type of this supplier is incorrect for the selected type'
+                                            ' of document.'))
+                if tax_payer_type in ['1', '3']:
+                    if latam_document_type_code in ['70', '71']:
+                        raise ValidationError(_('The tax payer type of this supplier is not entitled to deliver '
+                                                'fees documents'))
+                    if latam_document_type_code in ['110', '111', '112']:
+                        raise ValidationError(_('The tax payer type of this supplier is not entitled to deliver '
+                                                'imports documents'))
+                if tax_payer_type == '4' or country_id != self.env.ref('base.cl'):
+                    raise ValidationError(_('You need a journal without the use of documents for foreign '
+                                            'suppliers'))
+            if rec.journal_id.type == 'purchase' and not rec.journal_id.l10n_latam_use_documents:
+                if tax_payer_type != '4':
+                    raise ValidationError(_('This supplier should be defined as foreigner tax payer type and '
+                                            'the country should be different from Chile to register purchases.'))
```

Comment:
@ren-odoo These are quite some constraints. Would not it be more logical to apply them at the validation instead of every time?  We risk blocking the user, while when we do it on the validation of the invoice, he can still easily change it on the partner and go back.

Reply:
In this commit I am sending, the scope of the constraint is limited to posted entries.

456a7fcd06f151b6175fb4126c1bf02a060b733a